konig-schemagen/src/main/java/io/konig/schemagen/sql/SqlTableGenerator.java

```diff
@@ -45,11 +48,43 @@ public SqlTableGenerator(SqlDatatypeMapper datatypeMapper) {
 	}
 	
 	public SqlTable generateTable(Shape shape) throws SchemaGeneratorException {
-		
+		
+		Shape shapeObj = new Shape();
 		String tableName = nameFactory.getTableName(shape);
-		SqlTable table = new SqlTable(tableName);
-		for (PropertyConstraint p : shape.getProperty()) {
-			SqlColumn column = column(shape, p);
+		SqlTable table = new SqlTable(tableName);
+		boolean reqflag = false;
+		if(shape.getOr()!=null){
+			reqflag = true;
+			List<Shape> shapeList = shape.getOr().getShapes();
+			List<PropertyConstraint> propertyConstraintList = new ArrayList<PropertyConstraint>();
+			for(int i =0; i<shapeList.size(); i++){
+				for (PropertyConstraint p : shapeList.get(i).getProperty()) {
+					if(propertyConstraintList.size()>0)
+					{
+						int count =0;
+						for(int j = 0; j<propertyConstraintList.size(); j++){
+							
+							if((propertyConstraintList.get(j).getPredicate().toString().equals(p.getPredicate().toString()))){
+								 count++;
+							}
+						}
+						if(count==0){
+							 p.setRequired(false);
```

Comment:
You should not be modifying the original property constraint.  You need to clone it and modify the minCount value on the clone.  Also, the logic for determining whether a property is required is not correct.